plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java

```diff
@@ -622,4 +622,28 @@ public static void showNodeInNavigator(DBNDatabaseNode dsNode) {
         }
     }
 
+    private static void removeUnrelatedMenuItems(Menu menu, DBNNode node) {
+        for (MenuItem item : menu.getItems()) {
+            Object itemData = item.getData();
+            if (itemData instanceof IContributionItem) {
+                IContributionItem contribution = (IContributionItem) itemData;
+                String id = contribution.getId();
+                if (id == null) {
+                    continue;
+                }
+                if (id.startsWith("org.eclipse.debug") || // $NON-NLS-0$
+                    id.startsWith("team.main") || // $NON-NLS-0$
+                    id.startsWith("addFromHistoryAction")) { // $NON-NLS-0$
+                    item.dispose();
+                }
```

Comment:
This logic breaks incapsulation. dbvfs resources internals has nothing to do with navigator.
Let's use instanceof DBNNodeWithResource. And add new function isRemoteResource there.
All dbfs resources are from DBNPathBase